test/tests/functional/pbs_schedule_indirect_resources.py

```diff
@@ -0,0 +1,123 @@
+        self.scheduler.add_resource(res)
+
+        # Set resource as the node_group_key
+        attr = {'node_group_enable': 'True', 'node_group_key': res}
+        self.server.manager(MGR_CMD_SET, SERVER, attr)
+
+    def submit_job(self, attr):
+        """
+        Helper function to submit a sleep job with provided attributes
+        """
+        job = Job(TEST_USER1, attr)
+        jobid = self.server.submit(job)
+
+        return (jobid, job)
+
+    def test_node_grouping_with_indirect_res(self):
+        """
+        Test node grouping with indirect resources set on some nodes
+        Steps:
+        -> Configure the system to have 6 vnodes with custom resource 'foostr'
+        set as the node_group_key
+        -> Set 'foostr' to 'A', 'B', and 'C' on the first three vnodes
+        -> Set 'foostr' for the next three vnodes to point to the first three
+        vnodes correspondingly
+        -> Verify that the last three vnodes are part of their respective
+        placement sets
+        """
+
+        # Create 6 vnodes
+        attr = {'resources_available.ncpus': 1}
+        self.server.create_vnodes('vnode', attr, 6, mom=self.mom)
+
+        # Configure a system with 6 vnodes and 'foostr' as node_group_key
+        self.config_complex_for_grouping('foostr')
+
+        # Set 'foostr' to 'A', 'B' and 'C' respectively for the first
+        # three vnodes
+        attr = {'resources_available.foostr': 'A'}
+        self.server.manager(MGR_CMD_SET, NODE, attr, 'vnode[0]')
+        attr = {'resources_available.foostr': 'B'}
+        self.server.manager(MGR_CMD_SET, NODE, attr, 'vnode[1]')
+        attr = {'resources_available.foostr': 'C'}
+        self.server.manager(MGR_CMD_SET, NODE, attr, 'vnode[2]')
+
+        # Set 'foostr' for last three vnodes as indirect resource to
+        # the first three vnodes correcpondingly
+        attr = {'resources_available.foostr': '@vnode[0]'}
+        self.server.manager(MGR_CMD_SET, NODE, attr, 'vnode[3]')
+        attr = {'resources_available.foostr': '@vnode[1]'}
+        self.server.manager(MGR_CMD_SET, NODE, attr, 'vnode[4]')
+        attr = {'resources_available.foostr': '@vnode[2]'}
+        self.server.manager(MGR_CMD_SET, NODE, attr, 'vnode[5]')
+
+        # Submit 3 jobs requesting 2 vnodes and check they ran on the nodes
+        # within same group
+        attr = {'Resource_List.select': '2:ncpus=1'}
+        j = [
+            self.submit_job(attr),
+            self.submit_job(attr),
+            self.submit_job(attr)]
+
+        # since there are 6 vnodes and the test grouped them on foostr
+        # resource, we now have groups in pair of vnode 0+3, 1+4, 2+5
+        # check that the jobs are running on correct vnode groups.
+        for i in range(3):
+            self.server.expect(JOB, {'job_state': 'R'}, id=j[i][0])
+            self.server.status(JOB, 'exec_vnode', j[i][0])
+            vn = j[i][1].get_vnodes()
```

Comment:
this might be easier to read if you just added "jid, j = self.submit_job(attr)" in the for loop, and then referenced jid and j in the loop.